File: exercises/RB101_small_problems/easy_9/welcome_stranger.rb

def greetings(name, details)
  name = name.join(' ')
  details = "#{details[:title]} #{details[:occupation]}"
  "Hello, #{name}! Nice to have a #{details} around."
end

puts greetings(['John', 'Q', 'Doe'], { title: 'Master', occupation: 'Plumber' })
# => Hello, John Q Doe! Nice to have a Master Plumber around.

File: JS_210/05_Objects/welcome_stranger.js
/*

Write a function that takes two arguments, an array and an object.
The array will contain two or more elements that, when combined with spaces,
produce a person's full name. The object will contain two keys, title and
occupation, and suitable values for both items.

Your function should log a greeting to the console that uses the person's
full name, and mentions the person's title and occupation.

*/

function greetings(nameArray, { title, occupation }) {
  let fullName = nameArray.join(' ');
  console.log(`Hello, ${fullName}! Nice to have a ${title} ${occupation} around.`)
}

greetings(['John', 'Q', 'Doe'], { title: 'Master', occupation: 'Plumber' });

// console output
// Hello, John Q Doe! Nice to have a Master Plumber around.